CRM/Contact/BAO/Query.php

```diff
@@ -4972,9 +4884,8 @@ public function searchQuery(
 
     $dao = CRM_Core_DAO::executeQuery($query);
 
-    if ($disableFullGroupByMode) {
-      CRM_Core_DAO::reenableFullGroupByMode();
-    }
+    // We can always call this - it will only re-enable if it was originally enabled.
+    CRM_Core_DAO::reenableFullGroupByMode();
```

Comment:
@monishdeb are you happy with this change based on the handling in the re-enable function being dependent on it having been enabled in the first place?